packages/utils/src/dal/mikro-orm/integration-tests/__tests__/mikro-orm-repository.spec.ts

```diff
@@ -0,0 +1,844 @@
+        where: { id: "1" },
+        options: { populate: ["entity2"] },
+      })
+
+      expect(listedEntities).toHaveLength(1)
+      expect(listedEntities[0]).toEqual(
+        expect.objectContaining({
+          id: "1",
+          title: "en1",
+        })
+      )
+      expect(listedEntities[0].entity2.getItems()).toHaveLength(0)
+    })
+
+    it("should successfully create an entity with a sub-entity one-to-many relation", async () => {
+      const entity1 = {
+        id: "1",
+        title: "en1",
+        entity2: [{ title: "en2-1" }, { title: "en2-2" }],
+      }
+
+      await manager1().upsertWithReplace([entity1], {
+        relationsToUpsert: ["entity2"],
+        relationsToSkip: [],
+      })
+      const listedEntities = await manager1().find({
+        where: { id: "1" },
+        options: { populate: ["entity2"] },
+      })
+
+      expect(listedEntities).toHaveLength(1)
+      expect(listedEntities[0]).toEqual(
+        expect.objectContaining({
+          id: "1",
+          title: "en1",
+        })
+      )
+      expect(listedEntities[0].entity2.getItems()).toHaveLength(2)
+      expect(listedEntities[0].entity2.getItems()).toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({
+            title: "en2-1",
+          }),
+          expect.objectContaining({
+            title: "en2-2",
+          }),
+        ])
+      )
+    })
+
+    it("should only update the parent entity of a one-to-many if relation is not included", async () => {
+      const entity1 = {
+        id: "1",
+        title: "en1",
+        entity2: [{ title: "en2-1" }, { title: "en2-2" }],
+      }
+
+      await manager1().upsertWithReplace([entity1], {
+        relationsToUpsert: ["entity2"],
+        relationsToSkip: [],
+      })
+      entity1.entity2.push({ title: "en2-3" })
+      await manager1().upsertWithReplace([entity1], {
+        relationsToUpsert: [],
+        relationsToSkip: ["entity2"],
+      })
+
+      const listedEntities = await manager1().find({
+        where: { id: "1" },
+        options: { populate: ["entity2"] },
+      })
+
+      expect(listedEntities).toHaveLength(1)
+      expect(listedEntities[0]).toEqual(
+        expect.objectContaining({
+          id: "1",
+          title: "en1",
+        })
+      )
+      expect(listedEntities[0].entity2.getItems()).toHaveLength(2)
+      expect(listedEntities[0].entity2.getItems()).toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({
+            title: "en2-1",
+          }),
+          expect.objectContaining({
+            title: "en2-2",
+          }),
+        ])
+      )
+    })
+
+    it("should successfully update, create, and delete subentities an entity with a one-to-many relation", async () => {
+      const entity1 = {
+        id: "1",
+        title: "en1",
+        entity2: [
+          { id: "2", title: "en2-1" },
+          { id: "3", title: "en2-2" },
```

Comment:
based on the above guard case, I would expect here for an update to happen, but it is indeed creating it. 

Example, if an ID is passed of a resource that is already deleted, it would recreate it instead of throwing an error.

Reply:
Same as above, for now we allow it, if we don't we can also potentially get read of a DB SELECT operation in the upsert